internal/plugin/connectors/http/generic/config_test.go

```diff
@@ -52,4 +58,54 @@ func Test_newConfig(t *testing.T) {
 		assert.Equal(t, "Basic Sm9uYWg6c2VjcmV0", headers["Authorization"])
 	})
 }
-// TODO: Add validation tests
+
+func Test_validate(t *testing.T) {
+	testCases := []struct {
+		description string
+		creds       map[string][]byte
+		expErrStr   string
+	}{
+		{
+			description: "validates good credentials",
+			creds: map[string][]byte{
+				"username": []byte("Jonah"),
+				"password": []byte("secret"),
+			},
+			expErrStr: "",
+		},
+		{
+			description: "errors on missing required credential",
+			creds: map[string][]byte{
+				"password": []byte("secret"),
+			},
+			expErrStr: "missing required credential",
+		},
+		{
+			description: "errors on invalid credential",
+			creds: map[string][]byte{
+				"username": []byte("Jon:ah"),
+				"password": []byte("secret"),
+			},
+			expErrStr: "doesn't match pattern",
+		},
+	}
+
+	cfg, err := getSampleConfig(t)
+	if err != nil {
+		return
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.description, func(t *testing.T) {
+			err = cfg.validate(tc.creds)
+			if tc.expErrStr == "" {
+				assert.NoError(t, err)
+			} else {
+				assert.Error(t, err)
+				if err != nil {
+					assert.True(t, strings.Contains(err.Error(), tc.expErrStr))
+				}
+			}
+		})
+	}
```

Comment:
We try to follow [line of sight programming](https://youtu.be/5DVV36uqQ4E?t=660) wherever possible.  This almost always makes the code cleaner and  the logic easier to follow code because the happy path and the "exceptions," in the colloquial sense of that word, are easily differentiated.  It also reduces indentation level. In this case the change is almost trivial:
```go
	for _, tc := range testCases {
		t.Run(tc.description, func(t *testing.T) {
			err = cfg.validate(tc.creds)
			if tc.expErrStr == "" {
				assert.NoError(t, err)
				return
			}
			assert.Error(t, err)
			if err != nil {
				assert.True(t, strings.Contains(err.Error(), tc.expErrStr))
			}
		})
	}
```

Reply:
Fixed.